stock_dynamic_routing_delivery/models/stock_move.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2025 Camptocamp SA (https://www.camptocamp.com).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _get_new_picking_values(self):
+        # OVERRIDE to propagate the carrier, depending on the dynamic routing rule
+        # configuration.
+        vals = super()._get_new_picking_values()
+        if (
+            (routing_rule := self.env.context.get("__routing_rule"))
+            and routing_rule.propagate_carrier
+            and not vals.get("carrier_id")
+        ):
+            vals["carrier_id"] = self.group_id.sale_id.carrier_id.id
```

Comment:
I don't know how odoo is doing to propagate the carrier but shouldn't come from the sale. You better take it from the procurement group. There is an oca module adding it there.

Reply:
I tried to copy what Odoo does for their own `propagate_carrier`:
- https://github.com/odoo/odoo/blob/d737e896100cc5c7ffa62c7d13935001715c40f2/addons/stock_delivery/models/stock_move.py#L43

In STD there's no `carrier_id` in the procurement group, we'd need an extra dependency (`delivery_procurement_group_carrier`) but it makes no sense IMO

Reply:
Issue is that we do a carrier selection on the delivery. So we can't rely on the SO carrier.
I fixed the odoo propagation here https://github.com/OCA/stock-logistics-workflow/pull/2110